### PR TITLE
Jiemingy patch 6

### PR DESCRIPTION
**Pull request by Jiemingy**

### CODE REVIEW

File: Project-1/README.md [hunk 1]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
 3. Let V be the vocabulary of first-order logic consisting of a binary relation P and a unary relation F. Interpret P(x,y) as “x is a parent of y” and F(x) as “x is female.” Where possible define the following formulas in this vocabulary; where not possible, explain why: 
   
-  ```
-  (a)  B(x,y) that says that x is a brother of y  
-  (b)  A(x,y) that says that x is an aunt of y  
-  (c)  C(x,y) that says that x and y are cousins   
-  (d)  O(x) that says that x is an only child  
-  (e)  T(x) that says that x has exactly two brothers 
-  ```
+  (a)  B(x,y) ≡ ∃u(P(u,x)∧P(u,y))∧¬F(x)∧¬(x=y)
```
Comment (johnbeve) [inline Project-1/README.md:99]: Counterexample: u=x

File: Project-1/README.md [hunk 2]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
+
+  
+  (b)  A(x,y) ≡ ∃u∃w(P(u,y)∧P(w,x)∧P(w,u)∧¬(x=u))∧F(x)∧¬(x=y)
```
Comment (johnbeve) [inline Project-1/README.md:102]: Counterexample: Aunt by marriage.

File: Project-1/README.md [hunk 3]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
+  
+  
+  (c)  C(x,y) ≡ ∃u∃w∃z(P(w,z)∧P(w,u)∧P(z,x)∧P(u,y)∧¬(z=u))∧¬(x=y)
```
Comment (johnbeve) [inline Project-1/README.md:105]: Counterexample: Second cousins

File: Project-1/README.md [hunk 4]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
+    
+ 
+  (e)  T(x) ≡ ∃u∃y∃z(¬(y=z)∧¬(x=y)∧¬(x=z)∧P(u,x)∧P(u,y)∧P(u,z)∧¬F(y)∧¬F(z))∧∀w(P(u,w)∧¬F(w)→ w=x ∨ w=y ∨ w=z)
+  
 
 4. Let V be a vocabulary of the attribute (concept) language with complements (ALC) consisting of a role name "parent_of" and a concept name "Male". Interpret parent_of as "x is a parent of y" and M as "x is male". Where possible define the following formulas in this vocabulary; where not possible, explain why: 
   ```
   (a)  B that says that x is a brother of y
+
+  Answer:
```
Comment (johnbeve) [inline Project-1/README.md:118]: Should be "ALC is not expressive enough"

File: Project-1/README.md [hunk 5]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
+  Answer: 
+  
+  p3m (be a parent of exactly three male children) ≡ (≥3 parent_of. Male) ⊓ (≤3 parent_of. Male)
+  p1 (be a parent of at least one child) ≡ ≥1 parent_of. Person -- the inverse of p1 is just "be a child of"!
+  T ≡ ∃p3m¯.Person ⊔ (¬M ⊓ ∃p1¯. (≥2 parent_of. Male ⊓ ≤2 parent_of. Male))
 
 
 5. Select two formulas defined in ALC from question 4 to form the basis of a T-Box. Supplement this T-box with whatever other axioms you like, as well as an A-box, so that you ultimately construct a knowledge base K = (T,A). Provide a _model_ of K. This may be graphical or symbolic or both. 
 
+Answer:
```
Comment (johnbeve) [inline Project-1/README.md:160]: Should be "There are no formulas to choose from"

File: Project-1/README.md [hunk 6]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
 ∃x∀y∃z(R(x,y) ∧ R(x,z) ∧ R(y,z))
-```
+
+Answer: 
+
+The following graph models the second sentence but not the first sentence:
+Verticles: a, b, c
+Edges: ab, ac, aa (loop)
+
+See attached picture: image 0.
```
Comment (johnbeve) [inline Project-1/README.md:205]: Please express the graph in mermaid in the README. Do not include separate image files.

File: Project-1/README.md [hunk 7]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
+
 	
 8. Using an online tableau proof generator - such as the one found here `https://www.umsu.de/trees/` - provide tree proofs of the following entailments, which are known as the De Morgan's laws:
-  ```(a) ∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+  (a) ∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+  Tree proof:
+  ¬∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
```
Comment (johnbeve) [inline Project-1/README.md:211]: What justifies moving from ¬∀x∀y to ¬∀y without first removing the negation?

File: Project-1/README.md [hunk 8]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
 
+
+  first step: (¬p ∨ ¬q) → ¬(p ∧ q)
+   
+ 1. ¬p ∨ ¬q          premise
+|2. ¬¬(p ∧ q)        assumption
+|3. p ∧ q            ¬E2
+|4. p                ∧E3
+|5. q                ∧E3
+|6. ⊥                ⊥I 1,4,5
```
Comment (johnbeve) [inline Project-1/README.md:268]: Expand this

File: Project-1/README.md [hunk 9]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
+  (d) ∀x∀y(¬(Px ∨ Qx) → ¬(Px ∧ Qx))
+ Tree proof: same as (c)
+ 
 	
 9. Using a natural deduction proof generator - such as the one found here `https://proofs.openlogicproject.org/` - provide natural deduction proofs for each of De Morgan's laws. 
 
+
+  first step: (¬p ∨ ¬q) → ¬(p ∧ q)
+   
+ 1. ¬p ∨ ¬q          premise
```
Comment (johnbeve) [inline Project-1/README.md:263]: I asked you to prove quantified versions of Demorgan's Laws. You've attempted to prove propositional versions. You need to include the quantifiers.

File: Project-1/README.md [hunk 10]
```diff
@@ -12,63 +12,296 @@ Tip #4: The work we do is challenging; that should be assumed. You are smart eno
...
+See the attachment: image 3.
+
+Fourth step:  ¬p ∧ ¬q → ¬(p v q)
+
+see the attachment: image 4.
+
+
 10. Compare and contrast the proofs provided for (a) in your answers to questions 8 and 9. Explain the different assumptions, strategies, etc. exhibited in tree proofs vs natural deduction proofs. 
 
+For (a), the natural deduction proof applies mutiple deductive rules including disjunction elimination or double negation elimination (DNE), while the Tree proof only uses indirect derivation. That is a tree proof just focuses on finding possible logical contradictions among negated subformulas of the target formula.
```
Comment (johnbeve) [inline Project-1/README.md:305]: Tree proofs use deductive rules too, such as conjunction elimination and DeMorgan's laws! The main difference is natural deduction allows for more proof methods, e.g. conditional, than tableau which only uses indirect.

**Review by johnbeve — CHANGES_REQUESTED** [review]

File: Project-1/README.md (hunk 4 above)
Comment (Jiemingy) [reply Project-1/README.md:118]: Thank you for pointing it out1

**Pull request closed without merging**